internal/services/elasticsan/elastic_san_volume_snapshot_data_source_test.go

```diff
@@ -0,0 +1,189 @@
+// so we only onboard this as a data source instead of a resource. The acctest creates the snapshot as a test step
+func TestAccElasticSANVolumeSnapshotDataSource_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_elastic_san_volume_snapshot", "test")
+	d := ElasticSANVolumeSnapshotDataSource{}
+
+	data.DataSourceTestInSequence(t, []acceptance.TestStep{
+		{
+			Config: d.snapshotSource(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				data.CheckWithClientForResource(func(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) error {
+					if _, ok := ctx.Deadline(); !ok {
+						var cancel context.CancelFunc
+						ctx, cancel = context.WithTimeout(ctx, 30*time.Minute)
+						defer cancel()
+					}
+
+					volumeId, err := volumes.ParseVolumeID(state.ID)
+					if err != nil {
+						return err
+					}
+
+					id := snapshots.NewSnapshotID(volumeId.SubscriptionId, volumeId.ResourceGroupName, volumeId.ElasticSanName, volumeId.VolumeGroupName, data.RandomString)
+
+					snapshot := snapshots.Snapshot{
+						Properties: snapshots.SnapshotProperties{
+							CreationData: snapshots.SnapshotCreationData{
+								SourceId: volumeId.ID(),
+							},
+						},
+					}
+
+					client := clients.ElasticSan.Snapshots
+					if err = client.VolumeSnapshotsCreateThenPoll(ctx, id, snapshot); err != nil {
+						return fmt.Errorf("creating %s: %+v", id, err)
+					}
+
+					return nil
+				}, "azurerm_elastic_san_volume.test"),
+			),
+		},
+		{
+			Config: d.snapshotRestore(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("source_id").IsNotEmpty(),
+				check.That(data.ResourceName).Key("source_volume_size_in_gib").IsNotEmpty(),
+				check.That(data.ResourceName).Key("volume_name").IsNotEmpty(),
+			),
+		},
+		{
+			Config: d.snapshotSource(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				data.CheckWithClientForResource(func(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) error {
+					if _, ok := ctx.Deadline(); !ok {
+						var cancel context.CancelFunc
+						ctx, cancel = context.WithTimeout(ctx, 30*time.Minute)
+						defer cancel()
+					}
+
+					volumeId, err := volumes.ParseVolumeID(state.ID)
+					if err != nil {
+						return err
+					}
+
+					id := snapshots.NewSnapshotID(volumeId.SubscriptionId, volumeId.ResourceGroupName, volumeId.ElasticSanName, volumeId.VolumeGroupName, data.RandomString)
+
+					client := clients.ElasticSan.Snapshots
+					if err = client.VolumeSnapshotsDeleteThenPoll(ctx, id); err != nil {
+						return fmt.Errorf("creating %s: %+v", id, err)
+					}
+
+					return nil
+				}, "azurerm_elastic_san_volume.test"),
+			),
+		},
+	})
+}
+
+func (d ElasticSANVolumeSnapshotDataSource) snapshotSource(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+variable "primary_location" {
+  default = %q
+}
+variable "random_integer" {
+  default = %d
+}
+variable "random_string" {
+  default = %q
+}
```

Comment:
This looks like a remnant from a resource/data source generated with Pandora, please remove these
```suggestion
```

Reply:
removed